src/services.py

```diff
@@ -0,0 +1,153 @@
+from PIL import Image, ImageFont, ImageDraw
+from bs4 import BeautifulSoup as bs
+import requests as req
+import textwrap
+
+class ImageGenerator:
+    fontDirectory = "src/resources/fonts/RobotoBlack.ttf"
+    storyTemplateDirectory = "src/resources/templates/storyTemplate.jpg"
+
+    def generateImage(self, contest):
+        MAX_W = 1080
+        font = ImageFont.truetype(self.fontDirectory, 50)
+        count = 0
+        image = Image.open(self.storyTemplateDirectory)
+        draw = ImageDraw.Draw(image)
+        w, h = font.getsize("Platform : " + contest.platform)
+        draw.text((((MAX_W - w) / 2), 900), "Platform : " +
+                  contest.platform, fill="white", font=font, align="center")
+        w, h = font.getsize("Contest code : " + contest.contestCode)
+        draw.text((((MAX_W - w) / 2), 1000), "Contest code : " +
+                  contest.contestCode, fill="white", font=font, align="center")
+        font = ImageFont.truetype(self.fontDirectory, 70)
+        lines = textwrap.wrap(contest.contestName, width=25)
+        y_text = 1150
+        for line in lines:
+            width, height = font.getsize(line)
+            draw.text(((MAX_W - width) / 2, y_text), line, font=font, fill="white")
+            y_text += height
+        font = ImageFont.truetype(self.fontDirectory, 50)
+        draw.text((150, y_text+100), "START : ",
+                  fill="white", font=font, align="left")
+        timeDifference = contest.endDateTime - contest.startDateTime
+        contest.startDateTime = str(
+            contest.startDateTime.strftime('%d-%m-%Y %H:%M:%S'))
+        contest.endDateTime = str(
+            contest.endDateTime.strftime('%d-%m-%Y %H:%M:%S'))
+        draw.text((400, y_text+100), contest.startDateTime,
+                  fill="white", font=font, align="left")
+        draw.text((150, y_text+200), "END : ", fill="white",
+                  font=font, align="right")
+        draw.text((400, y_text+200), contest.endDateTime,
+                  fill="white", font=font, align="right")
+        timeDifference = str(timeDifference)
+        duration = ""
+        print(timeDifference)
+        durationList = timeDifference
+        if "days" in timeDifference:
+            daysList = timeDifference.split(" days, ")
+            duration = daysList[0] + " days "
+        if "days" in timeDifference:
+            durationList = timeDifference.split(" days, ")
+            durationList = durationList[1]
+        #print(durationList)
+        durationList = durationList.split(":")
+        print("hi")
+        if(durationList[0] != "00"):
+            duration += str(durationList[0]) + " Hours "
+        if(durationList[1] != "00"):
+            duration += str(durationList[1]) + " Minutes "
+        if(durationList[2] != "00"):
+            duration += str(durationList[2]) + " Seconds"
+        w, h = font.getsize("DURATION : " + duration)
+        draw.text((((MAX_W - w) / 2), y_text+300), "DURATION : " +
```

Comment:
Can we set BG of an image?
if yes, please pick a few colors, and let's randomize the background color picking.